tests/js/client/shell/shell-index.js

```diff
@@ -112,9 +113,49 @@ function IndexSuite() {
       }
       assertTrue(seenProgress, "Never saw progress being reported!");
     },
+    
+    testIndexProgressIsNotReported : function () {
+      internal.debugSetFailAt("fillIndex::pause");
+      let idxdesc = { name: "progress", type: "persistent", fields: ["name"], inBackground: true};
+
+      let job = arango.POST_RAW(`/_api/index?collection=${cn}`, idxdesc,
+                                {"x-arango-async": "store"}).headers["x-arango-async-id"];
+      let count = 0;
+      // check for 5 seconds that the in-progress index does not appear in the result.
+      // using 5 elapsed seconds as the test's stop condition is not a guarantee that
+      // the in-flight index won't appear later, but we do not have a better stop
+      // condition for the test.
+      while (true) {
+        // do not use withHidden here!
+        let idxs = arango.GET(`/_api/index?collection=${cn}`).indexes;
+        assertTrue(1, idxs.length, idxs);
+        assertEqual("primary", idxs[0].type);
+        sleep(0.1);
+        if (++count > 50) {
+          break;
+        }
+      }
+            
+      internal.debugSetFailAt("fillIndex::unpause");
+      
+      count = 0;
+      // wait until index appears, at most 30 seconds
+      while (++count < 300) {
+        let idxs = arango.GET(`/_api/index?collection=${cn}`).indexes;
+        assertEqual("primary", idxs[0].type);
+        if (idxs.length === 2) {
+          assertEqual("progress", idxs[1].name);
+          break;
+        }
+        sleep(0.1);
+      }
+      assertTrue(count < 300, count);
```

Comment:
This will almost certainly be unstable in ASAN runs. I suggest that we have this timeout here much longer.

Reply:
Done, see below.